### ISSUE TITLE
Pressing enter sometimes adds newline instead of accepting the ghost autocomplete

### ISSUE THREAD

**Issue by sanket-bhalerao** [microsoft/vscode#174653]
Type: <b>Bug</b>

Pressing enter sometimes adds a newline instead of accepting the ghost autocomplete.

what I mean by ghost autocomplete
![image](https://user-images.githubusercontent.com/9264813/219549820-b622a077-47fd-42b5-b4a6-72318edcb0e1.png)

VS Code version: Code - Insiders 1.76.0-insider (dc619de8d9efdabd781fc5658d0051813c347055, 2023-02-16T05:23:44.671Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.82GB (3.92GB free)|
|Process Argv|--crash-reporter-id e5294b27-02c9-4438-b8f3-90acb1091b3b|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|15.1.0
atlascode|atl|3.0.2
vscode-markdownlint|Dav|0.49.0
vscode-eslint|dba|2.4.0
githistory|don|0.6.19
gitlens|eam|13.2.0
prettier-vscode|esb|9.10.4
output-colorizer|IBM|0.1.2
vscode-scss|mrm|0.10.0
data-workspace-vscode|ms-|0.4.0
mssql|ms-|1.17.1
sql-bindings-vscode|ms-|0.3.0
sql-database-projects-vscode|ms-|0.21.0
vscode-typescript-next|ms-|5.0.202302160
vscode-yaml|red|1.11.0
sonarlint-vscode|Son|3.14.0
code-spell-checker|str|2.16.0
vscode-stylelint|sty|1.2.4
vscodeintellicode|Vis|1.2.30
vscodeintellicode-completions|Vis|1.0.21

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
vsc_aacf:30263846
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30404738
cppdebug:30492333
vscaat:30438846
vsclangdf:30492506
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
pythonsymbol12:30651887
showlangstatbar:30659908
vscodeenable:30660116
pythonb192cf:30661257

```

</details>

**Comment by timagixe** [comment]
I also started to face this issue. It is so annoying! When autocomplete opens it not always highlights the first suggestion, so pressing Enter key on keyboard results in adding new line.

**Comment by hediet** [comment]
Can you provide reproduction steps?

**Comment by timagixe** [comment]
@hediet of course I can, here is a repo for a quick start: https://github.com/timagixe/vscode-autocomplete-reproduction

Here is STR:
1. type `var`
2. press `Enter`
3. autosuggestions with colors opens
4. none of the colors is pre-selected
5. press `Enter` and this results in adding new line

I am also attaching a screencast of reproduction.


https://user-images.githubusercontent.com/60817786/220438891-d5a36f77-97a5-4a59-beec-c4112d9a4d50.mp4

**Comment by timagixe** [comment]
This is also happens in other file types, not only in css. Once I spot the exact reproduction steps I will share them here.

**Comment by outranker** [comment]
> @hediet of course I can, here is a repo for a quick start: https://github.com/timagixe/vscode-autocomplete-reproduction
> 
> Here is STR:
> 
> 1. type `var`
> 2. press `Enter`
> 3. autosuggestions with colors opens
> 4. none of the colors is pre-selected
> 5. press `Enter` and this results in adding new line
> 
> I am also attaching a screencast of reproduction.

This is exactly what I am experiencing as well. I feel like this bug got introduced in the latest version of vscode (version 1.75)

**Comment by Edchja** [comment]
I have the same issue. When autocomplete opens, pressing on tab to accept a suggestion will sometimes add a newline instead. The error occured mainly in .ts files

```
VS Code Version: 1.75.1
OS Version: Windows_NT x64 10.0.22621
```

**Comment by davcri** [comment]
Seems like a duplicate of https://github.com/microsoft/vscode/issues/173387 ?

**Comment by ksidibe** [comment]
This happens with PHP files too.
I've been having this issue for the last 2 or 3 weeks.
`Version: 1.76.2 (Universal) Commit: ee2b180d582a7f601fa6ecfdad8d9fd269ab1884`
`Date: 2023-03-14T17:54:09.061Z (1 wk ago)`
`Electron: 19.1.11`
`Chromium: 102.0.5005.196`
`Node.js: 16.14.2`
`V8: 10.2.154.26-electron.0`
`OS: Darwin x64 22.3.0 (Ventura 13.2.1)`

**Comment by chloezulcosky** [comment]
@sofiabe1 and I will be looking into this

**Comment by Archetipo95** [comment]
This is so annoying! It's happening even with the latest release...

**Comment by KwanghyunOn** [comment]
I have the same question too..

**Comment by rahmaniali-ir** [comment]
Same here too. It was working fine a while ago.

**Comment by Archetipo95** [comment]
`snippetsPreventQuickSuggestions` to false looks like is working :0

**Comment by rahmaniali-ir** [comment]
Thanks, @Archetipo95. Worked for me.

**Comment by chloezulcosky** [comment]
There is a setting "Editor › Suggest: Snippets Prevent Quick Suggestions" that controls whether an active snippet prevents quick suggestions. Copying the setting as JSON, it details that toggling this setting off causes "editor.suggest.snippetsPreventQuickSuggestions" to be set to false, which @Archetipo95 noted to be a fix. 

Therefore, when this setting is toggled 'off,' there should be the option to accept the ghost autocomplete instead of adding a new line when pressing 'enter.' [@sofiabe1]

**Comment by jrieken** [comment]
This got fixed when fixing https://github.com/microsoft/vscode/issues/173387

**Comment by Exalter104** [comment]
> editor.suggest.snippetsPreventQuickSuggestions

it not work for me i delete my keybinding enter key option mean accept suggestion word

**Comment by Edchja** [comment]
@Exalter104 on what VS Code version are you currently on? I'm on `1.79.2` and it should be fixed since the version `1.79.0`. 
If your VS Code version is lower than `1.79.XX` please update your VS Code🙂

After the update, you can remove the `editor.suggest.snippetsPreventQuickSuggestions` from the **settings.json** file or reset it back to the default value (`false`).

**Comment by sanket-bhalerao** [comment]
@jrieken, i think something weird is happening, as I start to type the unit test case `it` i see following
![image](https://github.com/microsoft/vscode/assets/9264813/4bdd0c02-846d-4f84-b9c2-b182d5ed321d)

if I press `tab` to accept the inline text the suggestion from content assist gets accepted.
i tried playing with `editor.suggest.snippetsPreventQuickSuggestions` setting but for both `true` and `false` this behavior remains same.
